src/style-spec/expression/parsing_context.js

```diff
@@ -61,20 +60,29 @@ class ParsingContext {
         index?: number,
         expectedType?: ?Type,
         bindings?: Array<[string, Expression]>,
-        options: {omitTypeAnnotations?: boolean} = {}
+        options: {typeAnnotation?: 'assert' | 'coerce' | 'omit'} = {}
     ): ?Expression {
         if (index) {
             return this.concat(index, expectedType, bindings)._parse(expr, options);
         }
         return this._parse(expr, options);
     }
 
-    _parse(expr: mixed, options: {omitTypeAnnotations?: boolean}): ?Expression {
-
+    _parse(expr: mixed, options: {typeAnnotation?: 'assert' | 'coerce' | 'omit'}): ?Expression {
         if (expr === null || typeof expr === 'string' || typeof expr === 'boolean' || typeof expr === 'number') {
             expr = ['literal', expr];
         }
 
+        function annotate(parsed, type, typeAnnotation: 'assert' | 'coerce' | 'omit') {
```

Comment:
I find `typeAnnotation` to be a very confusing name for this modifier (`typeModifier` 🤔 ) but we don't have to deal with that in this PR 😶